alacritty/src/config/mod.rs

```diff
@@ -43,29 +43,20 @@ pub enum Error {
 }
 
 impl std::error::Error for Error {
-    fn cause(&self) -> Option<&dyn (::std::error::Error)> {
-        match *self {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
             Error::NotFound => None,
-            Error::ReadingEnvHome(ref err) => Some(err),
-            Error::Io(ref err) => Some(err),
-            Error::Yaml(ref err) => Some(err),
-        }
-    }
-
-    fn description(&self) -> &str {
-        match *self {
-            Error::NotFound => "Couldn't locate config file",
-            Error::ReadingEnvHome(ref err) => err.description(),
-            Error::Io(ref err) => err.description(),
-            Error::Yaml(ref err) => err.description(),
+            Error::ReadingEnvHome(e) => e.source(),
```

Comment:
However I feel like it's just a style and I'd like to wait for @chrisduerr on this thing here. In our code base we do use `e` for such things, but way, way less then `err` as I can see.

Reply:
I'd agree with you if it was outside of a match - where I'd drop a lot more than just two characters if I could. I always leave edits from maintainers enabled, and I'd hope that would be exercised in a case where a non-functional style change holds up a pull. Or I can change it, either or - just trying to get rid of the build noise.

Reply:
If it's just a single variable and line length is not a problem, I think the longer the better.

With a short statement like this, even `error` would work perfectly fine, so that would resolve the issue of having to find the optimal abbreviation.

In general, `err` is the shortest 'acceptable' abbreviation usually though. Because `e` is commonly used for `event` too (`ev` would be better here!), so it's better not to have any reasons for confusion. Especially something like `e.source()` would make it impossible for me to tell if we're talking about an event or an error here, without looking at surrounding code.